### PR TITLE
Add PostgreSQL replication tests

### PR DESCRIPTION
**Pull request by mnagy**
This will fix (partially) #5841

@bparees @rhcarvalho fyi

### CODE REVIEW

**Comment by mnagy** [comment]
This should eventually be extended-tested with "core: images: postgresql". I won't trigger the tests now since the PostgreSQL image is broken.

**Comment by liggitt** [comment]
> I won't trigger the tests now since the PostgreSQL image is broken.

what is broken about the image?

**Comment by mfojtik** [comment]
@liggitt I guess https://github.com/openshift/postgresql/pull/81

**Comment by mnagy** [comment]
And https://github.com/openshift/postgresql/pull/80

**Comment by mnagy** [comment]
And https://github.com/openshift/postgresql/pull/78

**Comment by mnagy** [comment]
This should now be ready; PTAL

File: test/extended/util/db_image_helpers.go [hunk 1]
```diff
@@ -52,10 +58,26 @@ func NewMysql(c kclient.PodInterface, podName, masterPodName string) Database {
 	}
 }
 
+// NewPostgreSQL queries OpenShift for a pod with given name, saving environment
+// variables like username and password for easier use.
+func NewPostgreSQL(c kclient.PodInterface, podName, masterPodName string) Database {
```
Comment (bparees) [inline test/extended/util/db_image_helpers.go]: why are you passing PodInterface here? it's not used.

File: test/extended/util/db_image_helpers.go [hunk 2]
```diff
@@ -34,6 +34,12 @@ type MySQL struct {
 	MasterPodName string
 }
 
+// PostgreSQL is a PostgreSQL helper for executing commands
+type PostgreSQL struct {
```
Comment (bparees) [inline test/extended/util/db_image_helpers.go]: do you really need a separate struct for this?  how about renaming the MySQL struct to something more generic, assuming it's even necessary to have a wrapper struct for 2 string values...

File: test/extended/util/db_image_helpers.go (hunk 1 above)
Comment (bparees) [inline test/extended/util/db_image_helpers.go]: and same for NewMysql above.

File: test/extended/util/db_image_helpers.go (hunk 2 above)
Comment (bparees) [inline test/extended/util/db_image_helpers.go]: nm, I see why you created two structs, though you could still handle the fields jointly via composition.  but that's probably overkill.

File: test/extended/images/postgresql_replica.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+var (
+	postgreSQLReplicationTemplate = "https://raw.githubusercontent.com/openshift/postgresql/master/examples/replica/postgresql_replica.json"
+	postgreSQLEphemeralTemplate   = exutil.FixturePath("..", "..", "examples", "db-templates", "postgresql-ephemeral-template.json")
+	postgreSQLHelperName          = "postgresql-helper"
+	postgreSQLImages              = []string{
+		"openshift/postgresql-92-centos7",
+		"openshift/postgresql-94-centos7",
```
Comment (bparees) [inline test/extended/images/postgresql_replica.go]: what about rhel?

**Comment by mnagy** [comment]
PTAL

**Comment by mnagy** [comment]
[testonlyextended][extended:core(images)]

**Comment by bparees** [comment]
lgtm but failing tests.

**Comment by mnagy** [comment]
Looking. Looks like the helper PostgreSQL pod fails for some reason.

**Comment by mnagy** [comment]
I was wrong, it was the master that didn't go up. I upped timeouts.

**Comment by mnagy** [comment]
Wrong again, bad image names. [testonlyextended][extended:core(images: postgresql)]

**Comment by mnagy** [comment]
Found these errors:
```
$ cat container-postgresql-master-1-j5u7j-postgresql-master.log 
/usr/local/bin/run-postgresql.sh: line 26: exec: run-postgresql-master: not found
$ cat container-postgresql-slave-1-j8a2w-postgresql-slave.log 
Initializing PostgreSQL slave ...
pg_basebackup: directory "/var/lib/pgsql/data/userdata" exists but is not empty
```
These are bugs that were fixed. @sdodson, any idea? The images in question are:
 * `registry.access.redhat.com/openshift3/postgresql-92-rhel7`
 * `registry.access.redhat.com/rhscl/postgresql-94-rhel7`

**Comment by bparees** [comment]
postgres92 image hasn't been re-published yet.

for postges94 you'd have to ask @hhorak what the situation is.

**Comment by mnagy** [comment]
@hhorak the 9.4 seems to be missing this: https://github.com/openshift/postgresql/pull/82

**Comment by hhorak** [comment]
@mnagy Correct, it was merged just few days before GA release, so we couldn't include it any more. Is it something you need to fix asap? Then we need to track it in BZ: https://bugzilla.redhat.com/enter_bug.cgi?product=Red%20Hat%20Software%20Collections&component=rh-postgresql94-docker&version=rh-postgresql94 and need justification why it needs to be fixed.

**Comment by mnagy** [comment]
@bparees thoughts? I assume we want to have a fixed images published.

Meanwhile, I've updated the tests to exclude RHEL images for now.

**Comment by bparees** [comment]
the images are broken, they need to be patched and republished.  

@mnagy please open whatever bug the SCL team needs.

**Comment by mnagy** [comment]
@bparees will do. Can we merge this PR in the meantime so we at least cover centos images? I can open a new one that uncomments the rhel images that we can merge after fixed images are published.

**Comment by bparees** [comment]
yup, thanks @mnagy !

**Comment by bparees** [comment]
[merge]

**Pull request merged**